### PR TITLE
ONT-671 update governance contract

### PR DESCRIPTION
**Pull request by siovanus**

### CODE REVIEW

File: smartcontract/service/native/auth/param.go [hunk 1]
```diff
@@ -320,7 +320,7 @@ func (this *WithdrawParam) Deserialize(rd io.Reader) error {
 type VerifyTokenParam struct {
 	ContractAddr []byte
 	Caller       []byte
-	Fn           []byte
+	Fn           string
```
Comment (Honglei-Cong) [inline smartcontract/service/native/auth/param.go]: question: is this function name?

File: smartcontract/service/native/governance/param.go [hunk 2]
```diff
@@ -468,11 +498,76 @@ func (this *GlobalParam) Deserialize(r io.Reader) error {
...
+	this.Penalty = penalty
+	return nil
+}
+
+type SplitCurve struct {
+	Yi []uint64
+}
+
+func (this *SplitCurve) Serialize(w io.Writer) error {
+	if len(this.Yi) != 101 {
```
Comment (Honglei-Cong) [inline smartcontract/service/native/governance/param.go:531]: define 101 as constant.

File: smartcontract/service/native/auth/param.go (hunk 1 above)
Comment (siovanus) [reply smartcontract/service/native/auth/param.go]: yes, function name

File: smartcontract/service/native/governance/governance.go [hunk 3]
```diff
@@ -76,7 +78,8 @@ const (
 	SPLIT_CURVE     = "splitCurve"
 
 	//global
-	PRECISE = 1000000
+	PRECISE     = 1000000
+	ONG_PRECISE = 1000000000
```
Comment (laizy) [inline smartcontract/service/native/governance/governance.go]: define this based on ONG_DECIMIALS

File: smartcontract/service/native/governance/governance.go [hunk 4]
```diff
@@ -1660,16 +1667,17 @@ func depositTotalStake(native *native.NativeService, contract common.Address, ad
...
 	preStake := totalStake.Stake
-	//preTimestamp := totalStake.Timestamp
-	//
-	//amount := GetUnboundedOng(preStake, preTimestamp, native.Time)
-	//err = appCallTransferFromOng(native, utils.GovernanceContractAddress, utils.OntContractAddress, totalStake.Address, amount)
-	//if err != nil {
-	//	return errors.NewDetailErr(err, errors.ErrNoCode, "appCallTransferFromOng, transfer from ong error!")
-	//}
+	preTimestamp := totalStake.Timestamp
+	Timestamp := native.Time - constants.GENESIS_BLOCK_TIMESTAMP
```
Comment (laizy) [inline smartcontract/service/native/governance/governance.go]: rename timestamp to timeoffset to semantically make it more clear

**Review by tanZiWen — DISMISSED** [review]

**Review by laizy — APPROVED** [review]

**Pull request merged**